controller/src/main/java/io/pravega/controller/store/stream/Stream.java

```diff
@@ -352,22 +344,22 @@
      * Returns the latest stream epoch.
      * @return latest stream epoch.
      */
-    CompletableFuture<Pair<Integer, List<Integer>>> getLatestEpoch();
+    CompletableFuture<Pair<Integer, List<Long>>> getLatestEpoch();
```

Comment:
Surely this JavaDoc needs to change. If Txn segments can span multiple epocs and be committed in one, this method should explain when they are included and where. Same below.

Reply:
Transactions are created against a specific epoch. They live their until they are committed. And it is guaranteed that they will be committed in either the same epoch or a duplicate of the epoch. 
I will update the java doc as part of actual rolling txn implementation. 
Until it is implemented, its status quo as far as all these methods are concerned and their semantics dont change other than the fact that instead of using an int based segment number its a long based segment id.